src/MALA.jl

```diff
@@ -0,0 +1,103 @@
+using ForwardDiff: gradient!
+using DiffResults: GradientResult, value, gradient
+
+struct MALA{D} <: MHSampler
+    proposal::D
+end
+
+
+MALAMH(d) = MALA(RandomWalkProposal(d))
```

Comment:
We can just get rid of `MALAMH` and use `MALA` instead:

```suggestion
# Create a RandomWalkProposal if we weren't given one already.
MALA(d) = MALA(RandomWalkProposal(d))

# If we were given a RandomWalkProposal, just use that instead.
MALA(d::RandomWalkProposal) = MALA(d)
```

Reply:
That will lead to stackoverflow errors, I guess you want
```suggestion
# Create a RandomWalkProposal if we weren't given one already.
MALA(d) = MALA(RandomWalkProposal(d))

# If we were given a RandomWalkProposal, just use that instead.
MALA(d::RandomWalkProposal) = MALA{typeof(d)}(d)
```

Reply:
Good point -- I concur.